### PR TITLE
Add Laravel Pint

### PR DESCRIPTION
**Pull request by freddyheppell**
## Overview of Changes
* Adds Laravel Pint
  * Can be run locally with `sail bin pint`
* Check Pint in a separate CI check
  * This is because github's allow failure feature on actions is totally useless, there's not a good way to let the linting step fail the job but keep going
* Also realised a lot of the restructuring means Larastan wasn't checking code any more, added the new dirs and fixed.

## Deployment Requirements
- [ ] Composer update
- [ ] NPM update
- [ ] Database migration
- [ ] .env update
- [ ] Environment upgrade (eg. PHP version)
- [ ] Other (please specify)

### CODE REVIEW

File: app/Http/Controllers/LogoutController.php [hunk 1]
```diff
@@ -11,7 +11,8 @@ final class LogoutController extends WebController
 {
     public function __construct(
         private LogoutUseCase $logoutUseCase,
-    ) {}
+    ) {
+    }
```
Comment (andyksaw) [inline app/Http/Controllers/LogoutController.php:15]: Is there a way to customise the rule for this?

It's just a personal nitpick, but I feel like the brackets being on two lines is kind of ugly 😅

File: domain/BuilderRankApplications/Exceptions/ApplicationAlreadyInProgressException.php [hunk 2]
```diff
@@ -2,4 +2,6 @@
 
 namespace Domain\BuilderRankApplications\Exceptions;
 
-final class ApplicationAlreadyInProgressException extends \Exception {}
+final class ApplicationAlreadyInProgressException extends \Exception
+{
+}
```
Comment (andyksaw) [inline domain/BuilderRankApplications/Exceptions/ApplicationAlreadyInProgressException.php:7]: Same thing here

**Review by andyksaw — APPROVED** [review]
LGTM! Left some comments about 2 of the lint rules, but if it's not possible feel free to merge

File: app/Http/Controllers/LogoutController.php (hunk 1 above)
Comment (freddyheppell) [reply app/Http/Controllers/LogoutController.php:15]: Unfortunately for these brace issues, the only option right now is to totally disable the [`braces` rule](https://mlocati.github.io/php-cs-fixer-configurator/#version:3.8|fixer:braces), however there's a fair few issues on the CS fixer repo about splitting this rule up and allowing same-line braces for empty constructs, so hopefully it'll be possible to revert to our previous code style soon.

File: tests/Unit/Domain/Donations/UseCases/DeactivateExpiredDonorPerksUseCaseTest.php [hunk 3]
```diff
@@ -19,9 +19,13 @@ final class DeactivateExpiredDonorPerksUseCaseTest extends TestCase
     use RefreshDatabase;
 
     private GroupsManager $groupsManager;
+
     private DonationPerkRepository $donationPerkRepository;
+
     private Group $donorGroup;
+
     private Account $account;
+
```
[the remark this answers is not in the document]
Comment (freddyheppell) [reply tests/Unit/Domain/Donations/UseCases/DeactivateExpiredDonorPerksUseCaseTest.php:28]: I think changing to `only_if_meta` for `const` and `property` works best, this removes the whitespace, unless a property/const has a docblock when it adds one line. Unfortunately this also removes the whitespace between the subjects of this rule - so no empty line between the traits and properties.

[the remark this answers is not in the document]
Comment (freddyheppell) [reply tests/Unit/Domain/Donations/UseCases/DeactivateExpiredDonorPerksUseCaseTest.php:28]: Nevermind, turns out setting it to `none` actually means no spaces between but a space after the whole group, which is what we want.

**Pull request merged**